src/Components/Web/src/JSComponents/JSComponentInterop.cs

```diff
@@ -93,8 +75,7 @@ public virtual int AddRootComponent(string identifier, string domElementSelector
         /// <summary>
         /// For framework use only.
         /// </summary>
-        [JSInvokable]
-        public void SetRootComponentParameters(int componentId, int parameterCount, byte[] parametersJsonUtf8)
+        protected internal void SetRootComponentParameters(int componentId, int parameterCount, JsonElement parametersJson, JsonSerializerOptions jsonOptions)
```

Comment:
The changes in this method are all because the parameters now arrive as a `JsonElement` instead of a `byte[]`, because JS interop can supply a `JsonElement` natively without us having to manually do anything at either end.